### PR TITLE
Adds Assembly classes to our Sling Model for future development work

### PR DESCRIPTION
**Pull request by benradey**

### CODE REVIEW

File: pantheon-bundle/src/main/java/com/redhat/pantheon/asciidoctor/extension/MetadataExtractorTreeProcessor.java [hunk 1]
```diff
@@ -136,18 +137,24 @@ private void extractAbstract(List<StructuralNode> allNodes) {
      */
     private void extractModuleType(Document document) {
         Object attValue = document.getAttribute(MODULE_TYPE_ATT_NAME);
+        ModuleMetadata mm = null;
+        if (documentMetadata instanceof ModuleMetadata) {
```
Comment (carlosmunoz) [inline pantheon-bundle/src/main/java/com/redhat/pantheon/asciidoctor/extension/MetadataExtractorTreeProcessor.java:141]: I dislike `instanceof` 😢 
I wonder if extracting the module type is something we should externalize to a different asciidoctor extension. One that specializes in `Module`s. That way we build the extension set based on the type of document that we need to generate.

File: pantheon-bundle/src/main/java/com/redhat/pantheon/model/assembly/AssemblyVariant.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+    Child<AssemblyVersion> draft();
+
+    Child<AssemblyVersion> released();
+
+    @Override
+    AssemblyVariants getParent();
+
+    // TODO Not sure we need this
+    default AssemblyVersion getOrCreateVersion(String name) {
```
Comment (carlosmunoz) [inline pantheon-bundle/src/main/java/com/redhat/pantheon/model/assembly/AssemblyVariant.java:22]: I don't think we need this.

File: pantheon-bundle/src/main/java/com/redhat/pantheon/model/assembly/AssemblyVariant.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        return child(name, AssemblyVersion.class).get();
+    }
+
+    // TODO Not sure we need this
+    default AssemblyVersion createNextVersion() {
```
Comment (carlosmunoz) [inline pantheon-bundle/src/main/java/com/redhat/pantheon/model/assembly/AssemblyVariant.java:27]: I don't think we need this.

File: pantheon-bundle/src/main/java/com/redhat/pantheon/model/assembly/AssemblyVariants.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+    // The default variant
+    @Named(DEFAULT_VARIANT_NAME)
+    Child<AssemblyVariant> defaultVariant();
+
+    default Stream<AssemblyVariant> getVariants() {
+        return this.as(AssemblyVariant.class);
+    }
+
+    default AssemblyVariant getOrCreateVariant(String name) {
```
Comment (carlosmunoz) [inline pantheon-bundle/src/main/java/com/redhat/pantheon/model/assembly/AssemblyVariants.java:26]: this method is not needed either.

File: pantheon-bundle/src/main/java/com/redhat/pantheon/asciidoctor/extension/MetadataExtractorTreeProcessor.java (hunk 1 above)
Comment (benradey) [reply pantheon-bundle/src/main/java/com/redhat/pantheon/asciidoctor/extension/MetadataExtractorTreeProcessor.java:141]: Per our gchat conversation, I feel that this is a strength of the code, not a weakness. I would like to leave it as-is for now. We could revisit the decision later if things get more complicated.

File: pantheon-bundle/src/main/java/com/redhat/pantheon/model/assembly/AssemblyVariant.java (hunk 2 above)
Comment (benradey) [reply pantheon-bundle/src/main/java/com/redhat/pantheon/model/assembly/AssemblyVariant.java]: Removed.

File: pantheon-bundle/src/main/java/com/redhat/pantheon/model/assembly/AssemblyVariant.java (hunk 3 above)
Comment (benradey) [reply pantheon-bundle/src/main/java/com/redhat/pantheon/model/assembly/AssemblyVariant.java]: Removed.

File: pantheon-bundle/src/main/java/com/redhat/pantheon/model/assembly/AssemblyVariants.java (hunk 4 above)
Comment (benradey) [reply pantheon-bundle/src/main/java/com/redhat/pantheon/model/assembly/AssemblyVariants.java]: Removed.

**Pull request merged**